### PR TITLE
HTTPS alpha interface + stopgap implementation

### PR DESCRIPTION
**Pull request by antkmsft**
1. I think I do not understand spans completely yet, but currently there isn't enough time for me to figure it out. We probably shouldn't take both response_buffer and out_response as parameters.

2. There should now be powershell on mac and linux, so this might even work on other platforms, haha.

### CODE REVIEW

File: sdk/core/core/src/az_net.c [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+#else
+#include <unistd.h>
+#endif
+
+#include <_az_cfg_warn.h>
+
+#ifdef _MSC_VER
+// warning C4996: 'strcpy/strcat': This function or variable may be unsafe. Consider using
+// strcpy_s/strcat_s instead.
+#pragma warning(disable : 4996)
```
Comment (sergey-shandar) [inline sdk/core/core/src/az_net.c]: We should move it to `_az_cfg_warn.h`.

File: sdk/core/core/src/az_net.c [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+  Sleep(milliseconds);
+#else
+  usleep(milliseconds * 1000);
+#endif
+}
+
+static char const * alloc_shell_exec(char const * const cmd) {
+  assert(cmd != NULL);
+
+  FILE * const cmd_output = _popen(cmd, "r");
```
Comment (antkmsft) [inline sdk/core/core/src/az_net.c]: this one has no underscore on posix, make an ifdef

File: sdk/core/core/src/az_net.c [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+      = 4 * 1024; // trying to determine file size via fseek isn't going to work for the size
+                  // >=~1200 bytes for the reason that we are reading from a pipe. So we allocate
+                  // just large enough buffer and hope it is sufficient.
+
+  char * result = calloc(output_length, sizeof(char));
+  assert(result != NULL);
+
+  fgets(result, (int)output_length, cmd_output);
+
+  _pclose(cmd_output);
```
Comment (antkmsft) [inline sdk/core/core/src/az_net.c]: this one has no underscore in posix, it is only win32 that has

**Comment by antkmsft** [comment]
Will be obsolete soon

**Pull request closed without merging**